mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipperTest.java

```diff
@@ -0,0 +1,131 @@
+            .content(CONTENT_STREAM_1)
+            .size(SIZE_1)
+            .internalDate(SUN_SEP_9TH_2001)
+            .bodyStartOctet(0)
+            .flags(new Flags())
+            .propertyBuilder(new PropertyBuilder())
+            .mailboxId(TestId.of(1L))
+            .build();
+    private SimpleMailboxMessage MESSAGE_2 = SimpleMailboxMessage.builder()
+            .messageId(MESSAGE_ID_2)
+            .content(CONTENT_STREAM_2)
+            .size(SIZE_2)
+            .internalDate(SUN_SEP_9TH_2001)
+            .bodyStartOctet(0)
+            .flags(new Flags())
+            .propertyBuilder(new PropertyBuilder())
+            .mailboxId(TestId.of(1L))
+            .build();
+
+    private Zipper testee;
+    private File destination;
+
+    @BeforeEach
+    void beforeEach(TemporaryFolder temporaryFolder) throws Exception {
+        testee = new Zipper();
+        destination = File.createTempFile("backup-test", ".zip", temporaryFolder.getTempDir());
+    }
+
+    @Test
+    void archiveShouldWriteEmptyValidArchiveWhenNoMessage() throws Exception {
+        testee.archive(ImmutableList.of(), destination);
+
+        try (ZipFile zipFile = new ZipFile(destination)) {
+            assertThat(zipFile.getEntries().hasMoreElements()).isFalse();
+        }
+    }
+
+    @Test
+    void archiveShouldWriteOneMessageWhenOne() throws Exception {
+        testee.archive(ImmutableList.of(MESSAGE_1), destination);
+
+        try (ZipFile zipFile = new ZipFile(destination)) {
+            Enumeration<ZipArchiveEntry> entries = zipFile.getEntries();
+            assertThat(entries.hasMoreElements()).isTrue();
+            ZipArchiveEntry entry = entries.nextElement();
+            assertThat(entries.hasMoreElements()).isFalse();
+            assertThat(entry.getName()).isEqualTo(MESSAGE_ID_1.serialize());
+            assertThat(IOUtils.toString(zipFile.getInputStream(entry), MESSAGE_CHARSET)).isEqualTo(MESSAGE_CONTENT_1);
+        }
+    }
+
+    @Test
+    void archiveShouldWriteTwoMessagesWhenTwo() throws Exception {
+        testee.archive(ImmutableList.of(MESSAGE_1, MESSAGE_2), destination);
+
+        try (ZipFile zipFile = new ZipFile(destination)) {
+            Enumeration<ZipArchiveEntry> entries = zipFile.getEntries();
+            assertThat(entries.hasMoreElements()).isTrue();
+            ZipArchiveEntry entry1 = entries.nextElement();
+            assertThat(entries.hasMoreElements()).isTrue();
+            ZipArchiveEntry entry2 = entries.nextElement();
+            assertThat(entries.hasMoreElements()).isFalse();
+            assertThat(entry1.getName()).isEqualTo(MESSAGE_ID_1.serialize());
+            assertThat(IOUtils.toString(zipFile.getInputStream(entry1), MESSAGE_CHARSET)).isEqualTo(MESSAGE_CONTENT_1);
+            assertThat(entry2.getName()).isEqualTo(MESSAGE_ID_2.serialize());
+            assertThat(IOUtils.toString(zipFile.getInputStream(entry2), MESSAGE_CHARSET)).isEqualTo(MESSAGE_CONTENT_2);
```

Comment:
We gonna have a lot of ZIP assertions, right?

These assertions are hard to read, maybe we should invest early into having good quality ZIP testing tools.

I propose writing AssertJ assertions for it:

```
try (ZipFile zipFile = new ZipFile(destination)) {
    assertThatZip(new ZipFile(destination))
         .containsExactly(ZipEntryTest.builder()
             .name(MESSAGE_ID_1.serialize())
             .contentAsString(MESSAGE_CONTENT_1),
         ZipEntryTest.builder()
             .name(MESSAGE_ID_2.serialize())
             .contentAsString(MESSAGE_CONTENT_2));
}
```

That would make tests waaaay easier to read!

Reply:
Or better, we can write a wrapper on top of Zip to be more java 8 compliant ?

Reply:
I think we won't need all that stuff when we'll write restore methods.